MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/CreateCapillaryPressureModel.cpp

```diff
@@ -30,56 +28,76 @@ namespace PorousMedium
                   including <type>BrookCorey</type>
                   and it has a tag of <capillary_pressure>
 */
-static std::unique_ptr<CapillaryPressureSaturation>
-                         createBrookCorey(BaseLib::ConfigTree const& config)
+static std::unique_ptr<CapillaryPressureSaturation> createBrookCorey(
+    BaseLib::ConfigTree const& config)
 {
-    std::array<double, 5> parameters = {
-        {//! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__pd}
-         config.getConfigParameter<double>("pd"),
-         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__sr}
-         config.getConfigParameter<double>("sr"),
-         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__smax}
-         config.getConfigParameter<double>("smax"),
-         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__m}
-         config.getConfigParameter<double>("m"),
-         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__pc_max}
-         config.getConfigParameter<double>("pc_max")}};
-
-    assert(parameters[3] >= 1.0);  // m >= 1
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type}
+    config.checkConfigParameter("type", "BrookCorey");
+
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__pd}
```

Comment:
You're right. The line must be
`//! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__BrookCorey__pd}`
same for the other doxygen comments.
IMHO since documenting input file parameters is somewhat tricky, it could be fixed when writing documentation. But in order to write documentation it is better to first have test case where this material model is acutally used.